config/envoyconfig/http_connection_manager.go

```diff
@@ -71,7 +72,38 @@ func (b *Builder) buildLocalReplyConfig(
 		Mappers: []*envoy_http_connection_manager.ResponseMapper{{
 			Filter: &envoy_config_accesslog_v3.AccessLogFilter{
 				FilterSpecifier: &envoy_config_accesslog_v3.AccessLogFilter_ResponseFlagFilter{
-					ResponseFlagFilter: &envoy_config_accesslog_v3.ResponseFlagFilter{},
+					ResponseFlagFilter: &envoy_config_accesslog_v3.ResponseFlagFilter{
+						Flags: []string{
```

Comment:
The fact that Envoy doesn't provide any sort of `NOT` operator here is definitely annoying.

Would we be better off using a [Metadata Filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/accesslog/v3/accesslog.proto#envoy-v3-api-msg-config-accesslog-v3-metadatafilter)? Could we set a bit of metadata in the ext_authz response that tells this filter not to override the response body?

Reply:
For reasons I don't understand, using the metadata filter ends up always matching regardless of what I set.

```go
	filter := &envoy_config_accesslog_v3.AccessLogFilter{
		FilterSpecifier: &envoy_config_accesslog_v3.AccessLogFilter_MetadataFilter{
			MetadataFilter: &envoy_config_accesslog_v3.MetadataFilter{
				Matcher: &envoy_type_matcher_v3.MetadataMatcher{
					Filter: "envoy.filters.http.ext_authz",
					Path: []*envoy_type_matcher_v3.MetadataMatcher_PathSegment{
						{
							Segment: &envoy_type_matcher_v3.MetadataMatcher_PathSegment_Key{
								Key: DynamicMetadataPomeriumServiceKey,
							},
						},
					},
					Value: &envoy_type_matcher_v3.ValueMatcher{
						MatchPattern: &envoy_type_matcher_v3.ValueMatcher_StringMatch{
							StringMatch: &envoy_type_matcher_v3.StringMatcher{
								MatchPattern: &envoy_type_matcher_v3.StringMatcher_Exact{
									Exact: "authorizexxx",
								},
							},
						},
					},
				},
			},
		},
	}
```

I expected that filter to match nothing, but it was still true.

Reply:
Possibly https://github.com/envoyproxy/envoy/issues/17802